ush/python/pygfs/task/aero_emissions.py

```diff
@@ -37,48 +38,427 @@ def __init__(self, config: Dict[str, Any]) -> None:
+        gbbepxfiles = [os.path.basename(fname[0]) for fname in config['data_in']['gbbepx']['copy']]
+        climofiles = [os.path.basename(fname[0]) for fname in config['data_in']['climo']['copy']]
+        n_persist = config['n_persist']
 
         localdict = AttrDict(
-            {'variable_used_repeatedly': local_variable}
+            {
+                "cdate": blend_start_date,
+                "nforecast_days": nforecast_hours // 24,
+                "workdir": self.task_config.DATA,
+                "current_date": self.task_config.PDY,
+                'config': config,
+                'emistype': config['emistype'],
+                'climofiles': climofiles,
+                'qfedfiles': qfedfiles,
+                'hfedfiles': hfedfiles,
+                'gbbepxfiles': gbbepxfiles,
+                'n_persist': n_persist
+            }
         )
 
         # Extend task_config with localdict
         self.task_config = AttrDict(**self.task_config, **localdict)
 
+    @logit(logger)
+    def initialize(self) -> None:
+        """
+        Initialize the work directory by copying all the common fix data
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        None
+        """
+        logger.info("Copy Static Data to run directory")
+
+        data_in = self.task_config.config.data_in
+        emistype = self.task_config.emistype
+
+        # Copy climatology files to run directory except for HFED
+        if emistype.lower() != 'hfed':
+            logger.info(
+                f"Copy HFED '{data_in.hfed}' data to run directory"
+            )
+        logger.info("Copy climatology data to run directory")
+        FileHandler(data_in.climo).sync()
+        logger.info(f"Copy {emistype} data to run directory")
+        FileHandler(data_in[emistype.lower()]).sync()
+
+    @logit(logger)
+    def run(self) -> None:
+        """
+        Run the AerosolEmissions task with the given parameters.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        None
+        """
+        config_dict = self.task_config['config']
+        emistype = self.task_config['emistype']
+        ratio = config_dict['ratio']
+        climfiles = self.task_config['climofiles']
+        coarsen_scale = config_dict['coarsen_scale']
+        out_var_dict = config_dict['output_var_map']
+        current_date = self.task_config['current_date']
+        n_persist = config_dict['n_persist']
+
+        emission_map = {'qfed': self.task_config['qfedfiles'],
+                        'gbbepx': self.task_config['gbbepxfiles'],
+                        'hfed': self.task_config['hfedfiles']}
+
+        if emistype.lower() != 'blended':
+            try:
+                basefile = emission_map[emistype.lower()]
+            except KeyError as err:
+                raise KeyError(f"FATAL ERROR: {emistype.lower()} is not a supported emission type, ABORT!") from err
```

Comment:
Following the change above, you no longer need the emission_map:
```suggestion
        if emistype.lower() != 'blended':
            try:
                basefile = self.task_config[emistype.lower()}
            except KeyError as err:
                raise KeyError(f"FATAL ERROR: {emistype.lower()} is not a supported emission type, ABORT!") from err
```

Reply:
Addressed in bc9e8f95e67